## Data organized at:
* **`/home/ach/stylight/model_data/training/Coats_Jackets` ...**
* **`/home/ach/stylight/model_data/validation/Coats_Jackets` ...**
* **`/home/ach/stylight/model_data/testing/Coats_Jackets` ...**

In [1]:
import os
import random
import pandas as pd
import numpy as np
from shutil import copyfile

In [4]:
data_dir=r'C:\Users\Hp\FP\Blink Detection\Dataset Blink'
categories=os.listdir(data_dir)
print(categories)
origin_subfolders=[f'{data_dir}/{folder}' for folder in os.listdir(closed)]

['.ipynb_checkpoints', 'Closed', 'Opened', 'rename.ipynb', 'resize.ipynb', 'split_organize_training_validation_testing_data_folders.ipynb', 'to_jpg.ipynb']


NameError: name 'closed' is not defined

In [ ]:
for sub_dir in origin_subfolders:
    print(sub_dir,':', len(os.listdir(sub_dir)))

/home/ach/stylight/data/data_1/Coats_Jackets : 1406
/home/ach/stylight/data/data_1/Blazers_Suits : 945
/home/ach/stylight/data/data_1/Pants : 685
/home/ach/stylight/data/data_1/Tops : 1826
/home/ach/stylight/data/data_1/Skirts : 353
/home/ach/stylight/data/data_1/Underware : 1013
/home/ach/stylight/data/data_1/Shoes : 1432
/home/ach/stylight/data/data_1/Dresses_Overalls : 713
/home/ach/stylight/data/data_1/Accessories : 4332


### Define Destination Subfolders
Have been previously created

In [ ]:
training_dir='/home/ach/stylight/model_data/training'
print(os.listdir(training_dir))
validation_dir='/home/ach/stylight/model_data/validation'
print(os.listdir(validation_dir))
testing_dir='/home/ach/stylight/model_data/testing'
print(os.listdir(testing_dir))

['Coats_Jackets', 'Blazers_Suits', 'Pants', 'Tops', 'Skirts', 'Underware', 'Shoes', 'Dresses_Overalls', 'Accessories']
['Coats_Jackets', 'Blazers_Suits', 'Pants', 'Tops', 'Skirts', 'Underware', 'Shoes', 'Dresses_Overalls', 'Accessories']
['Coats_Jackets', 'Blazers_Suits', 'Pants', 'Tops', 'Skirts', 'Underware', 'Shoes', 'Dresses_Overalls', 'Accessories']


Obtain the **paths** of the **Destination Folders** for: `training`,`validation`,`testing`

Store in **`training_subfolders`,`validation_subfolders`,`testing_subfolders`**

In [ ]:
ori_subf= skirt
training_length=310
training_set_Sk = shuffled_set[0:training_length]
print('training: ',len(training_set_Sk))
validation_set_Sk = shuffled_set[311:346]
print(len(validation_set))
print('validation: ',len(validation_set_Sk))
testing_set_Sk=shuffled_set[-8:]
print('testing: ',len(testing_set_Sk))
testing_set_Sk

In [ ]:
training_subfolders=[f'{training_dir}/{folder}' for folder in os.listdir(training_dir)]
[print(subfolder) for subfolder in training_subfolders]
validation_subfolders=[f'{validation_dir}/{folder}' for folder in os.listdir(validation_dir)]
print('\n')
[print(subfolder) for subfolder in validation_subfolders]
testing_subfolders=[f'{testing_dir}/{folder}' for folder in os.listdir(testing_dir)]
print('\n')
[print(subfolder) for subfolder in testing_subfolders];

/home/ach/stylight/model_data/training/Coats_Jackets
/home/ach/stylight/model_data/training/Blazers_Suits
/home/ach/stylight/model_data/training/Pants
/home/ach/stylight/model_data/training/Tops
/home/ach/stylight/model_data/training/Skirts
/home/ach/stylight/model_data/training/Underware
/home/ach/stylight/model_data/training/Shoes
/home/ach/stylight/model_data/training/Dresses_Overalls
/home/ach/stylight/model_data/training/Accessories


/home/ach/stylight/model_data/validation/Coats_Jackets
/home/ach/stylight/model_data/validation/Blazers_Suits
/home/ach/stylight/model_data/validation/Pants
/home/ach/stylight/model_data/validation/Tops
/home/ach/stylight/model_data/validation/Skirts
/home/ach/stylight/model_data/validation/Underware
/home/ach/stylight/model_data/validation/Shoes
/home/ach/stylight/model_data/validation/Dresses_Overalls
/home/ach/stylight/model_data/validation/Accessories


/home/ach/stylight/model_data/testing/Coats_Jackets
/home/ach/stylight/model_data/testing/Blaz

In [ ]:
import shutil

for ori_subf,cat in zip(origin_subfolders, categories):
    print(cat)
    nr_imgs=len(os.listdir(ori_subf))
    print(nr_imgs)
    imgs=os.listdir(ori_subf)
    #print(imgs[:2])
    shuffled_set = random.sample(imgs,nr_imgs)
    
    skirt='/home/ach/stylight/data/data_1/Skirts'
    if ori_subf != skirt:
        training_length=550
        training_set=shuffled_set[0:training_length]
        print(ori_subf,'- training:', len(training_set))
        for file in training_set:
            #shutil.copy2('/src/file.ext', '/dst/dir')
            shutil.copy2(f'{ori_subf}/{file}',f'/home/ach/stylight/model_data/training/{cat}')
            #print(f'{ori_subf}/{file}')

        validation_set = shuffled_set[551:661]
        print('validation: ',len(validation_set))
        for file in validation_set:
            shutil.copy2(f'{ori_subf}/{file}',f'/home/ach/stylight/model_data/validation/{cat}')
        
        testing_set=shuffled_set[661:]
        print('testing: ',len(testing_set))
        for file in testing_set:
            shutil.copy2(f'{ori_subf}/{file}',f'/home/ach/stylight/model_data/testing/{cat}')

Coats_Jackets
1406
/home/ach/stylight/data/data_1/Coats_Jackets - training: 550
validation:  110
testing:  745
Blazers_Suits
945
/home/ach/stylight/data/data_1/Blazers_Suits - training: 550
validation:  110
testing:  284
Pants
685
/home/ach/stylight/data/data_1/Pants - training: 550
validation:  110
testing:  24
Tops
1826
/home/ach/stylight/data/data_1/Tops - training: 550
validation:  110
testing:  1165
Skirts
353
Underware
1013
/home/ach/stylight/data/data_1/Underware - training: 550
validation:  110
testing:  352
Shoes
1432
/home/ach/stylight/data/data_1/Shoes - training: 550
validation:  110
testing:  771
Dresses_Overalls
713
/home/ach/stylight/data/data_1/Dresses_Overalls - training: 550
validation:  110
testing:  52
Accessories
4332
/home/ach/stylight/data/data_1/Accessories - training: 550
validation:  110
testing:  3671


Skirts has only 353 elements, take only 310 for traiing 45 for validation and 8 for testing

In [ ]:
orig_path='/home/ach/stylight/data/data_1/Skirts'
orig_files=os.listdir(orig_path)
training=orig_files[:-53]
for file in training:
    shutil.copy2(f'/home/ach/stylight/data/data_1/Skirts/{file}','/home/ach/stylight/model_data/training/Skirts')
validation=orig_files[-53:-8]
for file in validation:
    shutil.copy2(f'/home/ach/stylight/data/data_1/Skirts/{file}','/home/ach/stylight/model_data/validation/Skirts')

testing=orig_files[-8:]
for file in testing:
    shutil.copy2(f'/home/ach/stylight/data/data_1/Skirts/{file}','/home/ach/stylight/model_data/testing/Skirts')   